accord-core/src/main/java/accord/local/cfk/Pruning.java

```diff
@@ -393,23 +393,23 @@ private static Long2ObjectHashMap<TxnInfo> buildEpochPrunedBefores(TxnInfo[] byI
             return null;
 
         Long2ObjectHashMap<TxnInfo> epochPrunedBefores = new Long2ObjectHashMap<>();
-        for (long epoch = newPrunedBefore.epoch() ; epoch <= newPrunedBefore.executeAt.epoch(); ++epoch)
-            epochPrunedBefores.put(epoch, newPrunedBefore);
+        epochPrunedBefores.put(newPrunedBefore.epoch(), newPrunedBefore);
 
-        int maxi = Arrays.binarySearch(committedByExecuteAt, newPrunedBefore, TxnInfo::compareExecuteAt);
+        int maxi = -1 - Arrays.binarySearch(committedByExecuteAt, newPrunedBefore, (a, b) -> a.compareExecuteAtEpoch(b) >= 0 ? 1 : -1);
```

Comment:
do we need to check for < 0 here?

Reply:
The comparator never returns 0, so it is always a negative result.